torchdata/datapipes/iter/load/aisio.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from io import BytesIO
+from typing import Iterator, Tuple
+
+from torchdata.datapipes import functional_datapipe
+
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils import StreamWrapper
+
+try:
+    from aistore.client import Client
+    from aistore.pytorch.utils import parse_url, unparse_url
+
+    HAS_AIS = True
+except ImportError:
+    HAS_AIS = False
+
+
+def _assert_aistore() -> None:
+    if not HAS_AIS:
+        raise ModuleNotFoundError(
+            "Package `aistore` is required to be installed to use this datapipe."
+            "Please run `pip install aistore` or `conda install aistore` to install the package"
+            "For more info visit: https://github.com/NVIDIA/aistore/blob/master/sdk/python/"
+        )
+
+
+@functional_datapipe("list_files_by_ais")
+class AISFileListerIterDataPipe(IterDataPipe[str]):
+    """
+    Iterable Datapipe that lists files from the AIStore backends with the given URL prefixes. (functional name: ``list_files_by_ais``).
+    Acceptable prefixes include but not limited to - `ais://bucket-name`, `ais://bucket-name/`
+
+    Note:
+    -   This function also supports files from multiple backends (`aws://..`, `gcp://..`, `hdfs://..`, etc)
+    -   Input must be a list and direct URLs are not supported.
+    -   length is -1 by default, all calls to len() are invalid as
+        not all items are iterated at the start.
+    -   This internally uses AIStore Python SDK.
+
+    Args:
+        source_datapipe(IterDataPipe[str]): a DataPipe that contains URLs/URL
+                                            prefixes to objects on AIS
+        length(int): length of the datapipe
+        url(str): AIStore endpoint
```

Comment:
Do you mind change the order of these two arguments to the same order as the signature.